compiler/src/dotty/tools/dotc/typer/Namer.scala

```diff
@@ -530,28 +530,29 @@ class Namer { typer: Typer =>
    *  enter them into symbol table
    */
   def indexExpanded(origStat: Tree)(using Context): Context = {
-    def recur(stat: Tree): Context = stat match {
+    def recur(stat: Tree, idx: Int): Context = stat match {
       case pcl: PackageDef =>
         val pkg = createPackageSymbol(pcl.pid)
         index(pcl.stats)(using ctx.fresh.setOwner(pkg.moduleClass))
         invalidateCompanions(pkg, Trees.flatten(pcl.stats map expanded))
-        setDocstring(pkg, stat)
+        setDocstring(pkg, stat, idx)
         ctx
       case imp: Import =>
         ctx.importContext(imp, createSymbol(imp))
       case mdef: DefTree =>
         val sym = createSymbol(mdef)
         enterSymbol(sym)
-        setDocstring(sym, origStat)
+        setDocstring(sym, origStat, idx)
         addEnumConstants(mdef, sym)
         ctx
       case stats: Thicket =>
-        stats.toList.foreach(recur)
+        for (stat, i) <- stats.toList.zipWithIndex do
+          recur(stat, i)
         ctx
       case _ =>
         ctx
     }
-    recur(expanded(origStat))
+    recur(expanded(origStat), 0)
```

Comment:
This `idx` parameter looks really ugly and kludgy. Can't we instead add an `origin` parameter to `recur` which is usually the same as `origStat` but in the `Thicket`-`ExtMethods` case it's the appropriate element of `t.methods`?

Reply:
I changed it.

Reply:
That did not work, but I found a much simpler solution